src/executor.pas

```diff
@@ -3876,6 +3877,43 @@ procedure TExecutor.ExecCTable(ST: TCTableCommand);
 
 end;
 
+procedure TExecutor.ExecSurvival(ST: TCustomVariableCommand);
+var
+  Survival: TSurvival;
+  DF: TEpiDataFile;
+  Opt: TOption;
+  S: UTF8String;
+  VarNames: TStrings;
+begin
+  VarNames := ST.VariableList.GetIdentsAsList;
+  // Get the by variable if there is one
+  for Opt in ST.Options do
```

Comment:
Instead of looping through all possible options it is possible to use:
ST.Options.HasOption(..)
Which both checks and can return the options searched for. It is a lot cleaner to look at.

Reply:
Moved all Opt checks to the survival unit